sdks/java/harness/build.gradle

```diff
@@ -55,6 +55,8 @@ dependencies {
   compile library.java.joda_time
   compile library.java.slf4j_api
   compile library.java.vendored_grpc_1_13_1
+  // For executing KafkaIO, e.g. as an external transform
+  compile project(path: ":beam-sdks-java-io-kafka", configuration: "shadow")
```

Comment:
Note: `runtime` scope is sufficient.

Reply:
We might need a better way to make cross-language artifacts available. This also causes some problems with our build system.

```
FAILURE: Build failed with an exception.

* What went wrong:
Circular dependency between the following tasks:
:beam-runners-direct-java:buildDependents
\--- :beam-sdks-java-io-kafka:buildDependents
     \--- :beam-sdks-java-harness:buildDependents
          +--- :beam-runners-direct-java:buildDependents (*)
          \--- :beam-runners-java-fn-execution:buildDependents
               \--- :beam-runners-direct-java:buildDependents (*)
```

Reply:
I got a similar error while working on #8174. I've solved the problem by not packaging a required dependency together but just staging a dependency jar with #8340.

Reply:
Thanks for mentioning this. I was also thinking about an option to stage jars from Python. I think it would be nicer if the Jars were automatically provided for the user. The user has to download the KafkaIO jar now and supply its path?

The reason why we can't add KafkaIO to the SDK Harness is that Kafka depends on the direct-runner which has a dependency on the SDK Harness. I'm not sure why the direct-runner has to depend on the SDK harness.

Reply:
KafkaIO should have at most a test level dependency on the direct runner. Also the direct runner was meant to become the ULR and the interim the SDK harness was added so the direct runner would be able to launch and run Java pipelines "portably" without using a container. Now that the new environments exist, this shouldn't be needed anymore.

Reply:
@mxm I think it would be good if expansion service also provides additional dependencies when it responds to expansion requests. In case of Java, not only serialized objects themselves but also enclosing (or referenced) classes are needed for deserialization. For the completeness of serialized codes, they should be bundled together.

Reply:
@ihji The expansion service should return PTransforms that point to an environment with the needed jars.

Reply:
@lukecwik @ihji Could you have a look? I've added kafka-io to the SDK Harness container.

Reply:
I also think the expansion service should (in the future) have an option to specify additional dependencies that are not part of the environment.